### ISSUE TITLE
[Research] OUI Compliance audit of the `data` plugin

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/OpenSearch-Dashboards#4077]
Tracking issue for OUI compliance audit of the `data` plugin.

Follow the steps in #3945 to complete the audit, then open another issue with findings. Once the work of performing the audit is done, this issue can be closed.

If this issue isn't already assigned to someone and you'd like to take on this work, please ping @BSFishy in the comments and I can assign it to you.

**Comment by joshuarrrr** [comment]
@kavilla or @manasvinibs This would be a good choice to start with when you're ready to pick up an audit task.
